### ISSUE TITLE
[no-shadow.ignoreTypeValueShadow] false positive against imported types

### ISSUE THREAD

**Issue by MarkKahn** [typescript-eslint/typescript-eslint#3862]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
   "@typescript-eslint/no-shadow": [
      "error",
      { "ignoreTypeValueShadow": true }
    ]
  }
}
```

```TS
import type { foo } from './foo';
type bar = number;

// 'foo' is already declared in the upper scope
// 'bar' is fine
function doThing(foo: number, bar: number) {}
```

tsconfig:
```json
{
  "extends": "@tsconfig/node16/tsconfig.json",
  "compileOnSave": true,
  "compilerOptions": {
    "baseUrl": "./",
    "composite": true,
    "esModuleInterop": true,
    "exactOptionalPropertyTypes": true,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true,
    "importHelpers": true,
    "isolatedModules": true,
    "module": "ESNext",
    "moduleResolution": "node",
    "noEmit": true,
    "noImplicitAny": true,
    "noImplicitOverride": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noUncheckedIndexedAccess": true,
    "noUnusedLocals": true,
    "paths": {
      "@applications/*": ["src/applications/*"],
      "@modules/*": ["src/modules/*"],
      "package.json": ["package.json"]
    },
    "resolveJsonModule": true,
    "rootDirs": [".", "src/modules/seams"],
    "strict": true,
    "typeRoots": ["src/modules/@types", "node_modules/@types"]
  },
  "exclude": ["**/node_modules", "**/build", "**/dist"],
  "include": ["src/**/*", "./*", "./*.json", "config/*", "config/*.json"]
}
```

**Expected Result**

rule should ignore imported types just like it ignores types defined in the file

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.31.0` |
| `@typescript-eslint/parser`        | `4.31.0` |
| `TypeScript`                       | `4.4.2` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `16.8.0` |

**Comment by bradzacher** [comment]
This occurs because "type only" imports are actually treated as both a type and value by TS.
This is done so that `typeof foo` works properly. It's a little bit weird, I know.
The rule should have special-case handling for this, however.
